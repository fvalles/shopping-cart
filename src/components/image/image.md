Basic image component with default border='1px solid', border-color='productImageBorder', border-radius='4px' and margin='0px 16px 0px 0px':

```jsx
<Image height={70} width={140} />
```

Image component with custom properties:

```jsx
<Image
  height={80}
  width={100}
  border="3px dotted"
  borderColor="summaryBackground"
  borderRadius={8}
  margin="0"
/>
```
