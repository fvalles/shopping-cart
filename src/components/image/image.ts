import styled from 'styled-components';

export const Image = styled.img`
  margin-right: 16px;
  width: 72px;
  height: 72px;
  border: 1px solid #cacad1;
  border-radius: 4px;
`;
