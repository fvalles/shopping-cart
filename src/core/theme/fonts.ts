export const Fonts = {
  light: 'Avenir Light',
  medium: 'Avenir Medium',
  regular: 'Avenir Regular',
};

export type FontsType = typeof Fonts;
