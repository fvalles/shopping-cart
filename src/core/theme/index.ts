export { Colors } from './colors';
export { Fonts } from './fonts';
export { GlobalStyle } from './global-style';
