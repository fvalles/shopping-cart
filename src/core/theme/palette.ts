export const palette = {
  // color names are based on https://colornamer.netlify.app/
  black: '#000000',
  blackMarket: '#212240',
  blackMarketLight: '#21224029',
  bleachedSilk: '#F3F3F3',
  faintFern: '#DBDBE0',
  mischka: '#A6A7B3',
  sharpGrey: '#CACAD1',
  stormGrey: '#717285',
  venetianNights: '#7350FF',
  white: '#FFFFFF',
};
