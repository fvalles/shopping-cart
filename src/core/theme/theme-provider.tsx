import React, { FunctionComponent } from 'react';
import { DefaultTheme, ThemeProvider as Provider } from 'styled-components';
import { Colors, Fonts } from '.';

interface ThemeProviderProps {
  children?: React.ReactNode;
}

export const ThemeProvider: FunctionComponent<ThemeProviderProps> = ({
  children,
}) => (
  <Provider
    theme={
      {
        Colors,
        Fonts,
      } as DefaultTheme
    }>
    {children}
  </Provider>
);
