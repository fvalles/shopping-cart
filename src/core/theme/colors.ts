import { palette } from './palette';

export const Colors = {
  // color names are based on https://colornamer.netlify.app/
  primary: palette.venetianNights,
  productImageBorder: palette.sharpGrey,
  productPrice: palette.black,
  productQuantity: palette.faintFern,
  productTitle: palette.mischka,
  summaryBackground: palette.bleachedSilk,
  summaryHeading: palette.stormGrey,
  summaryText: palette.blackMarket,
  white: palette.white,
};

export type ColorsType = typeof Colors;
export type KeyColors = keyof ColorsType;
