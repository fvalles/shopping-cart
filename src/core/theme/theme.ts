import { DefaultTheme } from 'styled-components';
import { Colors } from './colors';

const theme: DefaultTheme = {
  Colors,
};

export { theme };
