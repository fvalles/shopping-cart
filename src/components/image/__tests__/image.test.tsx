import React from 'react';
import { ThemeProvider } from '@core/theme/theme-provider';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Image } from '../image';

describe('image component', () => {
  it('should render with default css properties', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Image alt="some image" height={72} src="imageSrc" width={72} />
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('border', '1px solid');
    expect(result).toHaveStyleRule('border-color', '#CACAD1');
    expect(result).toHaveStyleRule('border-radius', '4');
    expect(result).toHaveStyleRule('height', '72');
    expect(result).toHaveStyleRule('margin', '0px 16px 0px 0px');
    expect(result).toHaveStyleRule('width', '72');
    expect(result).toMatchSnapshot();
  });

  it('should render with custom css properties', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Image
            alt="some image"
            border="1px dashed"
            borderColor="primary"
            borderRadius={8}
            height={72}
            margin="20px"
            src="imageSrc"
            width={72}
          />
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('border', '1px dashed');
    expect(result).toHaveStyleRule('border-color', '#7350FF');
    expect(result).toHaveStyleRule('border-radius', '8');
    expect(result).toHaveStyleRule('height', '72');
    expect(result).toHaveStyleRule('margin', '20px');
    expect(result).toHaveStyleRule('width', '72');
    expect(result).toMatchSnapshot();
  });
});
