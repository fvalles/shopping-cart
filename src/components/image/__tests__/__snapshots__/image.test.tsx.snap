// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image component should render with custom css properties 1`] = `
.c0 {
  border: 1px dashed;
  border-color: #7350FF;
  border-radius: 8;
  height: 72;
  margin: 20px;
  width: 72;
}

<img
  alt="some image"
  className="c0"
  height={72}
  src="imageSrc"
  width={72}
/>
`;

exports[`image component should render with default css properties 1`] = `
.c0 {
  border: 1px solid;
  border-color: #CACAD1;
  border-radius: 4;
  height: 72;
  margin: 0px 16px 0px 0px;
  width: 72;
}

<img
  alt="some image"
  className="c0"
  height={72}
  src="imageSrc"
  width={72}
/>
`;
